### PR TITLE
chore: align active version terminology and link to docs

### PR DESCRIPTION
**Pull request by dannykopping**
I found this quite confusing when I came across this feature.

image

What is the "default version"? I've changed this to "active version" which I believe is the intent here, and added a link to the documentation.

image

I also linked the docs in the CLI `--require-active-version` flags' help text.

### CODE REVIEW

File: site/src/pages/TemplateVersionEditorPage/PublishTemplateVersionDialog.tsx [hunk 1]
```diff
@@ -10,11 +10,25 @@ import type { PublishVersionData } from "pages/TemplateVersionEditorPage/types";
...
+import { docs } from "../../utils/docs";
 
 export const Language = {
 	versionNameLabel: "Version name",
 	messagePlaceholder: "Write a short message about the changes you made...",
-	defaultCheckboxLabel: "Promote to default version",
+	defaultCheckboxLabel: "Promote to active version",
+	activeVersionHelpTitle: "Active versions",
+	activeVersionHelpText:
+		"Templates can enforce that a version be used for all workspaces (enterprise-only)",
```
Comment (johnstcn) [inline site/src/pages/TemplateVersionEditorPage/PublishTemplateVersionDialog.tsx:30]: This implies that any version can be enforced when it's only the active one.

```suggestion
		"Templates can enforce that the active version be used for all workspaces (enterprise-only)",
```

**Review by johnstcn — COMMENTED** [review]
Also need to update reference to "default version" in `docs/workspaces.md`

File: site/src/pages/TemplateVersionEditorPage/PublishTemplateVersionDialog.tsx [hunk 2]
```diff
@@ -10,11 +10,25 @@ import type { PublishVersionData } from "pages/TemplateVersionEditorPage/types";
...
+import { docs } from "../../utils/docs";
 
 export const Language = {
 	versionNameLabel: "Version name",
 	messagePlaceholder: "Write a short message about the changes you made...",
-	defaultCheckboxLabel: "Promote to default version",
+	defaultCheckboxLabel: "Promote to active version",
+	activeVersionHelpTitle: "Active versions",
+	activeVersionHelpText:
+		"Templates can enforce that the active version be used for all workspaces (enterprise-only)",
```
Comment (mafredri) [inline site/src/pages/TemplateVersionEditorPage/PublishTemplateVersionDialog.tsx:30]:
```suggestion
		"Templates can be configured to enforce the use of the active version across all workspaces (enterprise-only)",
```

**Review by mafredri — APPROVED** [review]
Nice!

File: site/src/pages/TemplateVersionEditorPage/PublishTemplateVersionDialog.tsx (hunk 2 above)
Comment (dannykopping) [reply site/src/pages/TemplateVersionEditorPage/PublishTemplateVersionDialog.tsx:30]: Feels more wordy to me.

**Review by johnstcn — COMMENTED** [review]
Post-merge 👍

File: site/src/pages/TemplateVersionEditorPage/PublishTemplateVersionDialog.tsx (hunk 2 above)
Comment (dannykopping) [reply site/src/pages/TemplateVersionEditorPage/PublishTemplateVersionDialog.tsx:30]: I hear you; I'm deferring to the linked documentation to fill in those details.

**Pull request merged**